compiler/rustc_typeck/src/check/mod.rs

```diff
@@ -482,6 +483,12 @@ fn typeck_with_fallback<'tcx>(
 
         fcx.select_all_obligations_or_error();
 
+        if !fcx.infcx.is_tainted_by_errors() {
+            fcx.check_transmutes();
+        }
+
+        fcx.check_asms();
```

Comment:
Why the difference between `check_transmutes` and `check_asms`?

Reply:
I wanted to replicate the previous diagnostics. For transmute that meant not to check if there were other errors already happening, as that would just cause useless errors in the transmute checks